src/events/events.service.ts

```diff
@@ -128,20 +134,29 @@ export class EventsService {
         hasPrevious: false,
       };
     }
-    const nextRecords = await this.prisma.event.findMany({
-      where,
-      orderBy,
-      cursor: { id: data[length - 1].id },
-      skip: 1,
-      take: 1,
-    });
-    const previousRecords = await this.prisma.event.findMany({
-      where,
-      orderBy,
-      cursor: { id: data[0].id },
-      skip: 1,
-      take: -1,
-    });
+
+    const [nextRecords = [], previousRecords = []] = await Promise.all([
```

Comment:
Can we remove the `Promise.all`? It's probably low risk here, but it can lead to db starvation. These queries should be fast anyways.

Reply:
makes sense